api/src/pcapi/core/educational/api/stock.py

```diff
@@ -29,11 +29,24 @@ def create_collective_stock(
 ) -> educational_models.CollectiveStock | None:
     offer_id = offer_id or stock_data.offer_id
     beginning = stock_data.beginning_datetime
+    start = stock_data.start_datetime
+    end = stock_data.end_datetime
     booking_limit_datetime = stock_data.booking_limit_datetime
     total_price = stock_data.total_price
     number_of_tickets = stock_data.number_of_tickets
     educational_price_detail = stock_data.educational_price_detail
 
+    assert beginning or (start and end)
+
+    if not beginning:
+        assert start
+        assert end
+        beginning = start
+
+    if not start or not end:
+        assert beginning
```

Comment:
idem, déjà testé par `assert beginning or (start and end)`, non ?

Reply:
ça passe mal avec mypy

Reply:
ok, parfois on écrit `  # helps mypy` à côté.

Reply:
c'est bien vu, j'ajoute ça.